File: java/ladder.java
import java.util.Scanner;

public class ladder {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int h = scan.nextInt();
		int v = scan.nextInt();
		System.out.println((int) Math.ceil(h / Math.sin(v * Math.PI / 180)));
		scan.close();
	}
}

File: ts/src/ladder.ts
import readline = require('readline')

function toRadians (degrees: number): number {
  return degrees * Math.PI / 180
}

function getLength (height: number, angle: number): number {
  return Math.ceil(height / Math.sin(toRadians(angle)))
}

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    const [height, degree] = line.split(' ').map(Number)
    console.log(getLength(height ?? 0, degree ?? 0))
  })
  .on('close', () => {
    process.exit(0)
  })
